src/main.rs

```diff
@@ -320,24 +411,46 @@ impl Kopium {
     }
 
     fn print_derives(&self, s: &Container) {
-        let mut derives = vec!["Serialize", "Deserialize", "Clone", "Debug"];
+        let mut derives: Vec<String> = ["Serialize", "Deserialize", "Clone", "Debug"]
+            .into_iter()
+            .map(String::from)
+            .collect();
+
         if s.is_main_container() && !self.hide_kube {
             // CustomResource first for root struct
-            derives.insert(0, "CustomResource");
+            derives.insert(0, "CustomResource".to_string());
         }
         if self.builders {
-            derives.push("TypedBuilder");
+            derives.push("TypedBuilder".to_string());
         }
-        // add user derives last in order
-        for d in &self.derive {
-            if s.is_enum && d == "Default" {
+
+        for derive in &self.derive {
+            if s.is_enum && derive.derived_trait == "Default" {
                 // Need to drop Default from enum as this cannot be derived.
                 // Enum defaults need to either be manually derived
                 // or we can insert enum defaults
                 continue;
             }
-            derives.push(d);
+
+            // Only insert the trait if the target matches our container.
+            if let Some(derived_trait) = match &derive.target {
+                DeriveTarget::All => Some(&derive.derived_trait),
+                DeriveTarget::Type(name) => (&s.name == name).then_some(&derive.derived_trait),
+                DeriveTarget::Structs => s.is_enum.then_some(&derive.derived_trait),
+                DeriveTarget::Enums { unit_only } => {
+                    if s.is_enum && (!unit_only || s.members.iter().all(|member| member.type_.is_empty())) {
```

Comment:
This possibly feels like it should be a helper on `Container` (but larger comment on that in main review body)